src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -541,31 +555,34 @@ private int WriteFrameToSendBuffer(MessageOpcode opcode, bool endOfMessage, Read
             {
                 // The server doesn't send a mask, so the mask offset returned by WriteHeader
                 // is actually the end of the header.
-                headerLength = WriteHeader(opcode, _sendBuffer, payloadBuffer, endOfMessage, useMask: false);
+                headerLength = WriteHeader(opcode, _sendBuffer, payloadBuffer, endOfMessage, useMask: false, compressed: _inflater is not null);
```

Comment:
shouldn't it be `_deflater is not null`, not `_inflater is not null`?

Reply:
100% :)